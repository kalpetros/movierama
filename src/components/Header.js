import React, { useContext } from 'react';

import { MoviesContext } from '../store/MoviesContext';

export const Header = () => {
  const { query, setQuery } = useContext(MoviesContext);

  const handleChange = (e) => {
    setQuery(e.currentTarget.value);
  };

  let title = 'In Theaters';
  if (query.length > 0) {
    title = `Movies "${query}"`;
  }

  return (
    <header className="header">
      <div className="header__content">
        <div>
          <div>
            <h1>Welcome</h1>
          </div>
          <div>
            <input
              type="text"
              value={query}
              className="input"
              placeholder="Type here to search for your favorite movie..."
              onChange={handleChange}
            ></input>
          </div>
        </div>
      </div>
      <div className="header__title">
        <div>
          <h1>{title}</h1>
        </div>
      </div>
    </header>
  );
};
