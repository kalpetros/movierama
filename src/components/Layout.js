import React from 'react';
import PropTypes from 'prop-types';

export const Layout = ({ children }) => {
  return (
    <main className="layout">
      <section>{children}</section>
    </main>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};
