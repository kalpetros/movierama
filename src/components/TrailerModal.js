import React, { useState } from 'react';
import { createPortal } from 'react-dom';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const TrailerModal = (props) => {
  const [visible, setVisible] = useState(false);
  const { id: id } = props;

  const handleClick = () => {
    setVisible((v) => (v ? false : true));
  };

  let portal = null;

  if (visible) {
    portal = createPortal(
      <div className="modal-wrapper" onClick={handleClick}>
        <div className="modal">
          <iframe
            width="1280"
            height="720"
            src={`https://www.youtube.com/embed/${id}`}
            frameBorder="0"
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
          ></iframe>
        </div>
      </div>,
      document.getElementById('modal')
    );
  }

  return (
    <div>
      <FontAwesomeIcon
        icon={['fab', 'youtube']}
        size="2x"
        color="red"
        className="button-icon"
        onClick={handleClick}
      />
      {portal}
    </div>
  );
};

TrailerModal.propTypes = {
  id: PropTypes.string.isRequired,
};
