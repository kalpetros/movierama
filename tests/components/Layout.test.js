import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { Layout } from '../../src/components/Layout';

describe('<Layout />', () => {
  let wrapper;

  beforeEach(() => {
    const children = <p>Test</p>;

    wrapper = shallow(<Layout>{children}</Layout>);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
