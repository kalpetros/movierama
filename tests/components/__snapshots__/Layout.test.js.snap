// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match the snapshot 1`] = `"<main class=\\"layout\\"><section><p>Test</p></section></main>"`;
